plugins/eos-blacklist/gs-plugin-eos-blacklist.c

```diff
@@ -207,17 +215,43 @@ app_get_flatpak_ref_name (GsApp *app)
 	return gs_app_get_metadata_item (app, "flatpak::RefName");
 }
 
+static gchar **
+get_default_locales (GsPlugin *plugin)
+{
+	g_auto(GStrv) flatpak_default_locales = NULL;
+	g_autoptr(GError) local_error = NULL;
+
+	GsPluginData *priv = gs_plugin_get_data (plugin);
+	if (priv->installation == NULL)
+		return NULL;
+
+	flatpak_default_locales = flatpak_installation_get_default_languages (priv->installation, &local_error);
+	if (local_error != NULL) {
+		g_warning ("No user locales could be retrieved! %s", local_error->message);
+		return NULL;
+	}
+
+	return g_steal_pointer (&flatpak_default_locales);
+}
+
 static gboolean
 gs_plugin_locale_is_compatible (GsPlugin *plugin,
 				const char *locale)
 {
 	g_auto(GStrv) locale_variants;
+	g_auto(GStrv) locale_options = NULL;
 	const char *plugin_locale = gs_plugin_get_locale (plugin);
 	int idx;
 
+	/* Check if a variant of a locale is compatible, e.g. 'pt' for 'pt_BR' */
 	locale_variants = g_get_locale_variants (plugin_locale);
-	for (idx = 0; locale_variants[idx] != NULL; idx++) {
-		if (g_strcmp0 (locale_variants[idx], locale) == 0)
+	if (locale_variants != NULL && g_strv_contains ((const gchar * const *) locale_variants, locale))
+		return TRUE;
+
+	/* check if the app's locale is listed in the languages key for the ostree repo file */
+	locale_options = get_default_locales (plugin);
+	for(idx = 0; locale_options[idx] != NULL; idx++) {
+		if (g_str_has_prefix (locale, (const gchar * const) locale_options[idx]))
```

Comment:
I think [Rob’s comment](https://github.com/endlessm/gnome-software/pull/482#pullrequestreview-295910306) still applies here. We can’t do prefix-matching on locales. I think you need a separate function to compare two locales (`locale` and `locale_options[idx]` or `locale_variants[idx]`) and make a decision based on that. The only way to reliably compare two locales is to parse them completely, following the format described in [`man 3 setlocale`](https://jlk.fjfi.cvut.cz/arch/manpages/man/setlocale.3), i.e. ` language[_territory][.codeset][@modifier]`. Each component can be compared using `g_strcmp0()`.

As I understand it, you want to check that at least one of the user’s locale settings (their current locale, or the results of `get_default_locales()`) is equal to, or less specific than, the `locale` describing the flatpak.

Happy to discuss this on Slack if that would help unblock things. :)